src/submission/submission.service.ts

```diff
@@ -32,22 +35,34 @@ export class SubmissionService extends BaseService<Submission> {
     return this.findAll(query);
   }
 
-  async sendSubmission(gameId: string, exerciseId: string, playerId: string, file: IFile): Promise<Submission> {
-    const submission: Submission = await super.create({
-      game: gameId,
-      player: playerId,
-      exerciseId: exerciseId,
-    } as Submission);
+  async sendSubmission(
+    gameId: string,
+    exerciseId: string,
+    challengeId: string,
```

Comment:
Problem is that an exercise can belong to many challenges. Why add challenge here?

Reply:
To query the ChallengeStatus by Player_id and Challenge_id, if the challenge is locked or hidden then we should not create a submission imo

Reply:
An exercise can belong to many challenges, that's why `sendSubmission` did not receive a `challengeId`. I suggest to process submission, but only complete a challenge if it's in OPENED or AVAILABLE state.